shared/utility/math/euler.hpp

```diff
@@ -168,6 +168,92 @@ namespace utility::math::euler {
         return quat.matrix();
     }
 
+    /**
+     * @brief Convert rotation matrix to euler angles Thetaxy = [Roll, Pitch, Yaw]
+     */
+    template <typename Scalar>
+    void rot2rpy(const Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>& Rxy,
+                 Eigen::Matrix<Scalar, Eigen::Dynamic, 1>& Thetaxy) {
+        assert(Rxy.rows() == 3);
+        assert(Rxy.cols() == 3);
+        Thetaxy.resize(3, 1);
+        Thetaxy << atan2(Rxy(2, 1), Rxy(2, 2)), atan2(-Rxy(2, 0), sqrt(pow(Rxy(2, 1), 2) + pow(Rxy(2, 2), 2))),
+            atan2(Rxy(1, 0), Rxy(0, 0));
+    }
+
+    /**
+     * @brief Convert rotation matrix to euler angles Thetaxy = [Roll, Pitch, Yaw]
+     */
+    template <typename Scalar>
+    Eigen::Matrix<Scalar, Eigen::Dynamic, 1> rot2rpy(const Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>& Rxy) {
+        assert(Rxy.rows() == 3);
+        assert(Rxy.cols() == 3);
+        Eigen::Matrix<Scalar, Eigen::Dynamic, 1> Thetaxy;
+        Thetaxy.resize(3, 1);
+        Thetaxy << std::atan2(Rxy(2, 1), Rxy(2, 2)),
+            std::atan2(-Rxy(2, 0), std::sqrt(pow(Rxy(2, 1), 2) + std::pow(Rxy(2, 2), 2))),
```

Comment:
I've got a similar function in my Odometry branch in this repo, might be good to check the differences.